### ISSUE TITLE
Embedded images

### ISSUE THREAD

**Issue by gusilveiramp** [nahid/imapx#6]
Hello Friend,

Congratulations for the work. I'm using on a project.
I have a question, see if you can help me:
I can not open e-mails with embedded images. E-mails with embedded images are displayed the following to me:
![img](https://cloud.githubusercontent.com/assets/5175759/16903884/fd872c50-4c60-11e6-8669-cb2f77585dc2.png)

I know this is base64_encode, but even if I use the base64_decode function, they continue to be displayed as follows:
![img2](https://cloud.githubusercontent.com/assets/5175759/16903887/028b9376-4c61-11e6-8aea-d1189342531c.png)

You know how it could be done?

**Comment by nahid** [comment]
Is its an attachments file/s?

**Comment by gusilveiramp** [comment]
No. Only embedded images (inserted in the email body).
Whenever there is an image embedded in the email body, the message is coded the way I showed you.

If the email is a common html email, open normally, for example, google advices, social media advices, etc. The problem is when I insert the image directly in the email body.
If I insert an image by embedding images gmail tool, hotmail or any other e-mail tool, it shows the encrypted email.

![step-2](https://cloud.githubusercontent.com/assets/5175759/16915452/efb3816a-4ccd-11e6-85e5-057b0dc14f48.png)
![step-3](https://cloud.githubusercontent.com/assets/5175759/16915454/efb7e278-4ccd-11e6-8d42-9a553c269f78.png)
![step-4](https://cloud.githubusercontent.com/assets/5175759/16915453/efb7883c-4ccd-11e6-8ed4-00c365ece989.png)
![step-7](https://cloud.githubusercontent.com/assets/5175759/16915457/efba693a-4ccd-11e6-9ed4-bd61e04077e4.png)
![step-5](https://cloud.githubusercontent.com/assets/5175759/16915455/efb80fbe-4ccd-11e6-81a0-d93e29b82253.png)
![step-6](https://cloud.githubusercontent.com/assets/5175759/16915456/efb9b710-4ccd-11e6-99c5-ca9e3547f720.png)

**Comment by nahid** [comment]
Here is my output with embedded image. Which method do you used for fetching email body?

![imap](https://cloud.githubusercontent.com/assets/3167309/16915581/1b21403e-4d1a-11e6-87c5-d088baf8c4f8.jpg)

**Comment by gusilveiramp** [comment]
Yes, with pinterest works perfectly here too. I update my last message. Please, see the image that I upload.
The problem is when you follow exactly the steps on my last message.

**Comment by gusilveiramp** [comment]
![step-8](https://cloud.githubusercontent.com/assets/5175759/16915794/e91356a8-4ccf-11e6-9c33-9de84f4e81d6.png)

This is a default html email I received from my Google+ network. It opens normally. The problem is when I create an email and manually insert the image in the email body using the tool to embed images. Exactly as I showed in the previous message.

**Comment by gusilveiramp** [comment]
Hey friend, here a similar problem in Stackoverflow:
http://stackoverflow.com/questions/17106488/embedded-images-causing-problems-with-imap-fetchbody

I'm working on it. Any news, I'll let you know.

**Comment by nahid** [comment]
Thank you :) . If you find any solution please let me know or make a pull request
